### PR TITLE
Aida 1158

### PR DESCRIPTION
**Pull request by sharknc**
- Created new hypothesis-max-size command line argument that controls the maximum size for a hypothesis file during validation
- Added tests to test new hypothesis-max-size cli argument
- Updated documentation for `--hypothesis-max-size` argument

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ Usage:  <br>
 |`--ont=FILE ...` | validate against the OWL-formatted ontolog(ies) at the specified filename(s) |
 |`--nist` | validate against the NIST restrictions |
 |`--nist-ta3` | validate against the NIST hypothesis restrictions (implies `--nist`) |
+|`--hypothesis-max-size=num` | Specify the maximum size of a hypothesis file in MB when validating against the NIST hypothesis restrictions (`--nist-ta3`). Default is 5 |
```
Comment (nextcen-dgemoets) [inline README.md]: It looks like picocli uses the convention of putting the parameter in angle brackets (e.g., `--hypothesis-max-size=<hypothesisMaxSize>`).  We should probably make the README match the programmatic usage info.

But I'm certainly open to discussion on the matter.

File: java/src/main/java/com/ncc/aif/ValidateAIFCli.java [hunk 2]
```diff
@@ -83,6 +83,9 @@
     private static final int DEFAULT_DEPTH = 50;
     private static final int MINIMUM_DEPTH = 1;
 
+    //Hypothesis
+    private static final String DEAFULT_HYPOTHESIS_SIZE = "5"; //MB
```
Comment (nextcen-dgemoets) [inline java/src/main/java/com/ncc/aif/ValidateAIFCli.java]: Typo `DEAFULT` throughout.

File: java/src/main/java/com/ncc/aif/ValidateAIFCli.java [hunk 3]
```diff
@@ -110,9 +113,30 @@
...
 
+    @Option(names = "--hypothesis-max-size", defaultValue = DEAFULT_HYPOTHESIS_SIZE, description = "The maximum size of a hypothesis file in MB, default is 5",
+            arity = "1", converter = HypothesisMaxSizeConverter.class)
+    private int hypothesisMaxSize;
+
+    private static class HypothesisMaxSizeConverter implements CommandLine.ITypeConverter<Integer> {
+        @Override
+        public Integer convert(String value) {
+            try {
+                Integer size = "".equals(value) ? Integer.valueOf(DEAFULT_HYPOTHESIS_SIZE) : Integer.parseInt(value);
```
Comment (nextcen-dgemoets) [inline java/src/main/java/com/ncc/aif/ValidateAIFCli.java]: Change to `Integer.parseInt()`.

File: java/src/test/java/com/ncc/aif/ValidateCLITest.java [hunk 4]
```diff
@@ -138,6 +138,24 @@ void correctDepthWithoutValue() {
...
+    class HypothesisMaxSizeArgument {
+        @Test
+        void hypothesisMaxSizeBadType() {
+            expectUsageError(ValidateAIFCli.ERR_BAD_ARGTYPE.replaceAll("%.", ""),
+                    "--ldc", "--hypothesis-max-size", "s", "-t", "2", "-f", "tmp.ttl");
+        }
+        @Test
+        void hypothesisMaxSizeNegative() {
+            expectUsageError(ValidateAIFCli.ERR_BAD_ARGTYPE.replaceAll("%.", ""),
+                    "--ldc", "--hypothesis-max-size", "s", "-10", "2", "-f", "tmp.ttl");
```
Comment (nextcen-dgemoets) [inline java/src/test/java/com/ncc/aif/ValidateCLITest.java]: It looks like you're sending `--hypothesis-max-size` both an `s` and a `-10` as arguments.  I think you just want to send the -10?

File: README.md (hunk 1 above)
Comment (sharknc) [reply README.md]: Updated.

File: java/src/main/java/com/ncc/aif/ValidateAIFCli.java (hunk 2 above)
Comment (sharknc) [reply java/src/main/java/com/ncc/aif/ValidateAIFCli.java]: Fixed.

File: java/src/main/java/com/ncc/aif/ValidateAIFCli.java (hunk 3 above)
Comment (sharknc) [reply java/src/main/java/com/ncc/aif/ValidateAIFCli.java]: Fixed.

File: java/src/test/java/com/ncc/aif/ValidateCLITest.java (hunk 4 above)
Comment (sharknc) [reply java/src/test/java/com/ncc/aif/ValidateCLITest.java]: Fixed.

**Pull request merged**